### ISSUE TITLE
Update/improve CI setup for C-PAC

### ISSUE THREAD

**Issue by sgiavasis** [FCP-INDI/C-PAC#788]

**Comment by shnizzedy** [comment]
While working on #1211, I've made some progress on this issue as well. [`.circleci/config.yml`](https://github.com/FCP-INDI/C-PAC/blob/c05d23e8676d6f8adba90c06bdc5713e0c58922e/.circleci/config.yml) splits `build`, `test`, and `heatmaps` into separate jobs from c05d23e and persists the image from `build` to subsequent steps. If we can pare [`dev/circleci_data/pipe-test_ci.yml`](https://github.com/FCP-INDI/C-PAC/blob/c05d23e8676d6f8adba90c06bdc5713e0c58922e/dev/circleci_data/pipe-test_ci.yml) down to something that'll run in under 5 hours (#1197?), Circle should be able to run some tests for us again.

**Comment by sgiavasis** [comment]
Getting a 5h>pipeline is the block here? I can throw together something.

**Comment by shnizzedy** [comment]
Yeah. We can chain them together too, so if we can get through a complicated, long-running pipeline in multiple less-than-5-hour steps, we'll be in pretty good shape

**Comment by sgiavasis** [comment]
Oh, perfect. Let me revisit my testing pipelines and get back to you.
